unlaunchio/client/factory.go

```diff
@@ -38,37 +37,55 @@ func NewUnlaunchClientFactory(SDKKey string, cfg *UnlaunchClientConfig) (*Unlaun
 }
 
 // Client ...
-func (f *UnlaunchFactory) Client() *UnlaunchClient {
+func (f *UnlaunchFactory) Client() ClientInterface {
 
 	// TODO: Create and pass HTTP client instead of sdkey key, host
 	// like eventsCount
 
-	eventsRecorder := api.NewHTTPEventsRecorder(
-		util.NewHTTPClient(f.sdkKey, f.cfg.Host, f.cfg.HTTPTimeout, f.logger),
-		"/api/v1/impressions",
-		f.cfg.MetricsFlushInterval,
-		f.cfg.MetricsQueueSize,
-		"impressions",
-		f.logger)
-
-	eventsCounts := api.NewEventsCountAggregator(
-		util.NewHTTPClient(f.sdkKey, f.cfg.Host, f.cfg.HTTPTimeout, f.logger),
-		"/api/v1/events",
-		f.cfg.MetricsFlushInterval,
-		f.cfg.MetricsQueueSize,
-		f.logger)
-
-	hc := util.NewHTTPClient(f.sdkKey, f.cfg.Host, f.cfg.HTTPTimeout, f.logger)
-
-	return &UnlaunchClient{
-		FeatureStore: service.NewHTTPFeatureStore(
-			hc,
-			f.cfg.PollingInterval,
-			f.logger),
-		eventsRecorder:        eventsRecorder,
-		eventsCountAggregator: eventsCounts,
-		logger:                f.logger,
-		evaluator:             engine.NewEvaluator(f.logger),
+	if f.cfg.OfflineMode {
+
+		f.logger.Info("offline mode ", f.cfg.OfflineMode)
+
+		return &OfflineClient{
+			FeatureStore: service.NewHTTPFeatureStore(
```

Comment:
You don't need `FeatureStore `. This can be removed. Implementation doesn't depend on it, neither the interface